// @sentry/node/esm/index.js
import { Integrations } from '@sentry/core';
export { Hub, SDK_VERSION, Scope, addBreadcrumb, addEventProcessor, addGlobalEventProcessor, addIntegration, captureCheckIn, captureEvent, captureException, captureMessage, close, configureScope, continueTrace, createTransport, extractTraceparentData, flush, functionToStringIntegration, getActiveSpan, getActiveTransaction, getClient, getCurrentHub, getCurrentScope, getGlobalScope, getHubFromCarrier, getIsolationScope, inboundFiltersIntegration, lastEventId, linkedErrorsIntegration, makeMain, metrics, requestDataIntegration, runWithAsyncContext, setContext, setCurrentClient, setExtra, setExtras, setMeasurement, setTag, setTags, setUser, spanStatusfromHttpCode, startActiveSpan, startInactiveSpan, startSpan, startSpanManual, startTransaction, trace, withIsolationScope, withMonitor, withScope } from '@sentry/core';
export { autoDiscoverNodePerformanceMonitoringIntegrations } from './tracing/index.js';
export { NodeClient } from './client.js';
export { makeNodeTransport } from './transports/http.js';
export { defaultIntegrations, defaultStackParser, getSentryRelease, init } from './sdk.js';
export { DEFAULT_USER_INCLUDES, addRequestDataToEvent, extractRequestData } from '@sentry/utils';
export { deepReadDirSync } from './utils.js';
import { createGetModuleFromFilename } from './module.js';
export { createGetModuleFromFilename } from './module.js';
export { enableAnrDetection } from './integrations/anr/legacy.js';
import * as handlers from './handlers.js';
export { handlers as Handlers };
import * as index from './integrations/index.js';
import * as integrations from './tracing/integrations.js';
export { hapiErrorPlugin } from './integrations/hapi/index.js';
import { instrumentCron } from './cron/cron.js';
import { instrumentNodeCron } from './cron/node-cron.js';
import { instrumentNodeSchedule } from './cron/node-schedule.js';

/**
 * @deprecated use `createGetModuleFromFilename` instead.
 */
const getModuleFromFilename = createGetModuleFromFilename();

const INTEGRATIONS = {
  // eslint-disable-next-line deprecation/deprecation
  ...Integrations,
  ...index,
  ...integrations,
};

/** Methods to instrument cron libraries for Sentry check-ins */
const cron = {
  instrumentCron,
  instrumentNodeCron,
  instrumentNodeSchedule,
};

export { INTEGRATIONS as Integrations, cron, getModuleFromFilename };
